pkg/icingaredis/client.go

```diff
@@ -63,14 +64,13 @@ type HPair struct {
 // HYield yields HPair field-value pairs for all fields in the hash stored at key.
 func (c *Client) HYield(ctx context.Context, key string) (<-chan HPair, <-chan error) {
 	pairs := make(chan HPair)
```

Comment:
Then I'd at least make this channel buffered so that the next HSCAN call isn't waiting for the result of the previous one being processed.